core/domain/exp_domain.py

```diff
@@ -2327,8 +2327,10 @@ def get_state_names_mapping(self, change_list):
 
         for change_dict in reversed(change_list):
             if change_dict['cmd'] == CMD_RENAME_STATE:
-                old_to_new_state_names[change_dict['old_state_name']] = (
-                    old_to_new_state_names.pop(change_dict['new_state_name']))
+                if change_dict['new_state_name'] in old_to_new_state_names:
```

Comment:
What happens if a state gets renamed two or more times in the same changelist? Please add a test for that, too.

Reply:
@seanlip There is a test for that.  Line 713 - addition and multiple renames.

Reply:
Ah yes, my bad. Looks good, thanks!